mesonbuild/wrap/wrap.py

```diff
@@ -525,8 +543,24 @@ def get_data(self, urlstring: str) -> T.Tuple[str, str]:
         elif WHITELIST_SUBDOMAIN in urlstring:
             raise WrapException(f'{urlstring} may be a WrapDB-impersonating URL')
         else:
+            headers = {}
+            is_http_protocol = url.scheme in ('http', 'https')
```

Comment:
Allowing netrc for http-only schemes is a bit risky, since there's no warning that that is happening and we effectively send the user's passwords unencrypted...

I notice that e.g. curl only reads netrc when you ask it to, so if it sends them over insecure connections then, well, "you asked for it".

Reply:
I agree doing it only for https is safer.

Reply:
What if there was a `wrap-file` key to opt into either netrc support and/or `http` support?

Reply:
Do we really need to support that? If you have a setup that requires a password but does not use https then isn't that pretty much broken by default? If you are inside a trusted network (even VPN) you don't need passwords and if you are not in a secure network https is pretty much a must.

Reply:
I guess someone could always request HTTP support if it was ever needed. Then a use-case could be debated. Assuming HTTPS seems fine for now.

Reply:
> What if there was a `wrap-file` key to opt into either netrc support and/or `http` support? Alternatively a `setup` flag.

I would say, let's keep that out of scope, and if there is someone complaining we can always add that key and/or use `--allow-insecure` argument we already have on subprojects/wrap subcommands.

Reply:
I think ftp is fine, because for ftp there's literally no other choice and if you have to use ftp you're essentially expecting to communicate the password.

My issue was specifically http without s, as I think that's a case where the user almost certainly could and should be doing better.